src/systemctl/systemctl.c

```diff
@@ -8576,8 +8576,10 @@ static int halt_main(void) {
         if (r < 0)
                 return r;
 
-        if (arg_when > 0)
-                return logind_schedule_shutdown();
+        /* Delayed shutdown requested, and was successful */
+        if (arg_when > 0 && logind_schedule_shutdown() == 0)
```

Comment:
not that it matters, but usually we consider all positive values success, hence doing a `>= 0` check (rather than `== 0`) is a tiny bit nicer, as it otherwise one might think that > 0 means something special here, if one is used to our coding style of doing return values....